redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,43 @@
+package redisxtest
+
+import (
+	"context"
+	"time"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// NewMockRedisClient sets up a mock redis cluster, client and sender
+// This should be called from TestMain since the miniredis instance created is locked
+func NewMockRedisClient(ctx context.Context, timeout time.Duration) (client redisx.Syncx, teardown func(), err error) {
+	redisCluster, err := miniredis.Run()
+	if err != nil {
+		return redisx.Syncx{}, func() {}, err
```

Comment:
nit: in an error return, we would want to return `nil` for non-error returns whenever possible. you cannot return `nil` for the `client`, but you certainly can return `nil` for the `teardown` (we don't want people using it to blindly `defer teardown()` without checking for errors first).